pkg/generators/builder/builder.go

```diff
@@ -161,8 +161,18 @@ func (b *BuilderPatternGenerator) generateSettersForType(sw *generator.SnippetWr
 			if b.isTypeEnabled(m.Type) {
 				sw.Do(setter.GenerateSetterForEmbeddedStruct(m, b.getWrapperType(m.Type)))
 			}
-		case m.Type.Kind == types.Pointer && m.Type.Elem.Kind == types.Builtin:
-			sw.Do(setter.GenerateSetterForPointerToBuiltinType(m))
+		case m.Type.Kind == types.Pointer:
+			pointerType := m.Type.Elem
+			switch pointerType.Kind {
+			case types.Builtin:
+				sw.Do(setter.GenerateSetterForPointerToBuiltinType(m))
+			case types.Struct:
+				if b.isTypeEnabled(pointerType) {
+					sw.Do(setter.GenerateSetterForPrimitiveType(m))
```

Comment:
Let's add a new test case that covers this builder.

Reply:
thanks!
The test case exposed an issue. For pointers to enabled Struct types, the input parameter is our "wrapper" type and we have to extract the embedded k8s type, just like for EmbeddedSlice and EmbeddedStruct.

So I added a method `GenerateSetterForEmbeddedPointer`

To test that this works, in Kanopy I enabled generationg for `ServiceReference` and the code generated is as exected:
https://github.com/10gen/kanopy/pull/261/files#diff-0034bb0d2f6e41070dd49d16578a2a526b7f623f2bf82e482683cef79a8ee455R125

Reply:
Good catch.